Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Implementation
# [ODS-6434]Rebuild Database Templates build -Skip already successful package creation from working branch

Below is a structured breakdown of each use case:

Use Case 1:
Scenario:

A new commit occurs in one of the three repositories (Ed-Fi-ODS, Ed-Fi-ODS-Implementation, or Ed-Fi-Extensions).
The workflow begins by updating the CodeGen package but fails before completing.
Expected Behavior:

In the first run, the workflow updates the CodeGen package version. However, the workflow fails after this update, preventing the completion of other package updates.
Example Workflow Run:

[Failed after updating CodeGen package](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Implementation/actions/runs/10517476403/job/29141657598).
Use Case 2:
Scenario:

The workflow is re-triggered after the failure in Use Case 1.
It should skip the CodeGen package update since it was already updated and continue updating the Extensions packages (TPDM, Sample, Homograph).
Expected Behavior:

The workflow should resume from the Extensions package update, bypassing the CodeGen package. However, it fails again after updating the Extensions packages.
Example Workflow Run:

[Failed after updating Extensions packages](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Implementation/actions/runs/10517520017/job/29141781325).
Use Case 3:
Scenario:

The workflow is re-triggered after the failure in Use Case 2.
It should skip both the CodeGen and Extensions package updates, resuming with updates to the EdFi.Ods.Minimal.Template, EdFi.Ods.Minimal.Template.PostgreSQL, EdFi.Ods.Populated.Template, and EdFi.Ods.Populated.Template.PostgreSQL packages.
Expected Behavior:

The workflow should skip the already updated CodeGen and Extensions packages, but it fails after attempting to update the Minimal and Populated templates.
Example Workflow Run:

[Failed after updating Minimal and Populated templates](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Implementation/actions/runs/10518959780/job/29145570132).
Use Case 4:
Scenario:

The workflow is re-triggered after the failure in Use Case 3.
It should skip the CodeGen, Extensions, and Minimal/Populated template updates and continue with the updates for EdFi.Ods.Minimal.Template.TPDM, EdFi.Ods.Minimal.Template.TPDM.PostgreSQL, EdFi.Ods.Populated.Template.TPDM, and EdFi.Ods.Populated.Template.TPDM.PostgreSQL packages.
Expected Behavior:

The workflow successfully completes by updating the TPDM-related templates without any failures.
Example Workflow Run:

[Successful completion after TPDM package updates](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Implementation/actions/runs/10519122224/job/29145994222).
Validation Steps:
Branch Consistency Across Repositories:
Ensure the working branch exists in all three repositories (Ed-Fi-ODS, Ed-Fi-ODS-Implementation, Ed-Fi-Extensions).
Check for Previous Workflow Failures:
Before starting the process, check if there was a failure in the last run. If so, determine whether any packages were updated successfully.
Handling New Commits:
If new commits are detected after a failed workflow, ensure the workflow resumes from where it left off by skipping already updated packages.
